pageserver/src/tenant/timeline.rs

```diff
@@ -4006,6 +4037,67 @@ impl Timeline {
     }
 }
 
+pub struct DiskUsageEvictionInfo {
+    /// Timeline's largest layer (remote or resident)
+    pub max_layer_size: Option<u64>,
+    /// Timeline's resident layers
+    pub resident_layers: Vec<LocalLayerInfoForDiskUsageEviction>,
+}
+
+pub struct LocalLayerInfoForDiskUsageEviction {
+    pub layer: Arc<dyn PersistentLayer>,
+    pub last_activity_ts: SystemTime,
+}
+
+impl std::fmt::Debug for LocalLayerInfoForDiskUsageEviction {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        // format the tv_sec, tv_nsec into rfc3339 in case someone is looking at it
```

Comment:
Off-topic question, isnt it enough to have second resolution for last access time?

Reply:
I think this is an opportunity to cut down memory usage/simplify things. Second resolution should be enough (or millisecond if we really want). I'll create separate issue.

Reply:
https://github.com/neondatabase/neon/issues/3906